### PR TITLE
Changed references of `pystiche.ops.*Operator` to `pystiche.loss.*Loss`

### PR DESCRIPTION
**Pull request by sakshi-06**
This PR attempts to solve #547. Part of which has been resolved by @pmeier in #550.

I have made the changes based on my understanding, but I'll appreciate it if you can take a first look at this PR and suggest any changes required. 

*Note:*

While trying to build the documentation locally using `tox -e docs-html`, got the following error:

```
ERROR: InvocationError for command /home/sakshikumari/github-projects/pystiche/.tox/docs-html/bin/sphinx-build -M html source build -n (exited with code -9 (SIGKILL)) (exited with code -9)
________________________________________________________ summary _________________________________________________________
ERROR: docs-html: commands failed
```

Because of these errors, I could not test the documentation locally. Does the CI build the docs for each PR?

Thank you

cc: @krshrimali (since he has been guiding with the PRs)

cc: @pmeier

### CODE REVIEW

**Comment by pmeier** [comment]
@sakshi-06 That is indeed weird. Try to replace 

https://github.com/pystiche/pystiche/blob/04271974d24f044af754e8f4ae797edba1ab1c9f/tox.ini#L117

with

```ini
sphinx build -nb html source build
```

and then run `PYSTICHE_PLOT_GALLERY=0 tox -re docs-html`.

(The `PYSTICHE_PLOT_GALLERY=0` disables the usage examples which take a long time to build and I already handled them in #550. The `-r` flag tells `tox` to recreate the environment in case it is corrupted. You can drop it for consecutive runs.)

You should see a lot of warnings. Some of them do not point to `pystiche` but rather to `torch` or `torchvision`. While we should also fix them, this is out of scope for this PR.

---

If you still see the error from before, could you paste the output of `tox --version` and `tox -re docs-html --notest` here?

File: docs/source/gist/loss.rst [hunk 1]
```diff
@@ -9,32 +9,32 @@ The identification of content and style are core elements of a Neural Style Tran
...
 :class:`~pystiche.ops.EncodingOperator` encodes it first.
 
 In total ``pystiche`` supports four archetypes:
 
 +-------------------------------------------------------+-----------------------------------------------------------------------+
-| :class:`~pystiche.ops.Operator`                       | Builtin examples                                                      |
+| :class:`~pystiche.loss.Loss`                          | Builtin examples                                                      |
 +=======================================================+=======================================================================+
-| :class:`~pystiche.ops.PixelRegularizationOperator`    | - :class:`~pystiche.ops.TotalVariationOperator` :cite:`MV2015`        |
+| :class:`~pystiche.ops.PixelRegularizationOperator`    |   - :class:`~pystiche.loss.TotalVariationLoss` :cite:`MV2015`         |
```
Comment (pmeier) [inline docs/source/gist/loss.rst:29]:
```suggestion
| :class:`~pystiche.ops.PixelRegularizationOperator`    | - :class:`~pystiche.loss.TotalVariationLoss` :cite:`MV2015`         |
```

File: docs/source/gist/loss.rst [hunk 2]
```diff
@@ -9,32 +9,32 @@ The identification of content and style are core elements of a Neural Style Tran
...
 +-------------------------------------------------------+-----------------------------------------------------------------------+
 | :class:`~pystiche.ops.EncodingRegularizationOperator` |                                                                       |
 +-------------------------------------------------------+-----------------------------------------------------------------------+
 | :class:`~pystiche.ops.PixelComparisonOperator`        |                                                                       |
 +-------------------------------------------------------+-----------------------------------------------------------------------+
-| :class:`~pystiche.ops.EncodingComparisonOperator`     | - :class:`~pystiche.ops.FeatureReconstructionOperator` :cite:`MV2015` |
-|                                                       | - :class:`~pystiche.ops.GramOperator` :cite:`GEB2016`                 |
-|                                                       | - :class:`~pystiche.ops.MRFOperator` :cite:`LW2016`                   |
+| :class:`~pystiche.ops.EncodingComparisonOperator`     | - :class:`~pystiche.loss.FeatureReconstructionLoss` :cite:`MV2015`    |
```
Comment (pmeier) [inline docs/source/gist/loss.rst:35]:
```suggestion
| :class:`~pystiche.ops.EncodingComparisonOperator`     | - :class:`~pystiche.loss.FeatureReconstructionLoss` :cite:`MV2015` |
```

**Review by pmeier — COMMENTED** [review]
Hey @sakshi-06, I've added a new workflow to the CI, so we now always check the references to avoid broken ones in the future. You can see I've removed some temporary ignores from the `scripts/check_doc_references.py` file thanks to your PR! 

I have some minor formatting comments inline, but otherwise this LGTM! Note that you will need to reformat all rows of the table to get it working. Sphinx is pretty nitpicky about that.

If you want to continue working on these (not in this PR), remove some more ignores from the file, run `tox -e tests-docs` and fix the warnings that are shown.

**Comment by sakshi-06** [comment]
Hi @pmeier ,

Thanks for helping me with this PR. I've made the required changes, looks like all the tests are passing now. :)

I also changed the description of this PR to make it look better, for future reference.

Thank you again, please let me know if there is anything else that needs to be done with this PR. I'll be looking forward to contributing more to pystiche.

**Review by pmeier — APPROVED** [review]
LGTM, thanks a lot @sakshi-06!

> I'll be looking forward to contributing more to pystiche.

Awesome to hear that. If you want to keep fixing docs, just follow the steps I outlined in https://github.com/pystiche/pystiche/pull/556#pullrequestreview-747288873

**Pull request merged**